scrapy/selector/unified.py

```diff
@@ -182,9 +182,9 @@ def re_first(self, regex):
     def extract(self):
         return [x.extract() for x in self]
 
-    def extract_first(self):
+    def extract_first(self, default=None):
         for x in self:
-            return x.extract()
+            return x.extract() or default
```

Comment:
It doesn't look good to me. It is returning the default value only if first matched selection is an empty string.

I think the default value should be used only when there aren't matches like in:

```python
for x in self:
    return x.extract()
else:
    return default
```